rewrite-gradle/src/main/java/org/openrewrite/gradle/trait/GradleDependency.java

```diff
@@ -0,0 +1,181 @@
+            return null;
+        }
+
+        // If we're here, then we have a gradle dependency
+        Object firstArg = arguments.get(0);
+
+        if (firstArg instanceof G.MapEntry) {
+            // Format: implementation group: "commons-lang":commons-lang:2.6"
+            Map<String, String> parts = new HashMap<>();
+            arguments
+                    .forEach(arg -> {
+                        G.MapEntry mapEntry = (G.MapEntry) arg;
+                        String key = mapEntry.getKey().toString();
+                        String value = mapEntry.getValue().toString();
+                        if (key.equals("group")) {
+                            parts.put("group", value);
+                        } else if (key.equals("name")) {
+                            parts.put("name", value);
+                        } else if (key.equals("version")) {
+                            parts.put("version", value);
+                        }
+                    });
+
+            return new ResolvedGroupArtifactVersion(parts.get("group"), parts.get("name"), parts.get("version"));
+        } else if (firstArg instanceof G.GString) {
+            G.GString gString = (G.GString) firstArg;
+            List<String> strings = gString.getStrings().stream()
+                    .flatMap(s -> s instanceof J.Literal ?
+                            Stream.of(((J.Literal) s).getValue().toString().split(":")) :
+                            Stream.of("${" + ((G.GString.Value) s).getTree().toString() + "}"))
+                    .collect(Collectors.toList());
+            String group = !strings.isEmpty() ? strings.get(0) : null;
+            String artifact = strings.size() > 1 ? strings.get(1) : null;
+            String version = strings.size() > 2 ? strings.get(2) : null;
+            return new ResolvedGroupArtifactVersion(group, artifact, version);
+        } else if (firstArg instanceof J.Literal &&
+                ((J.Literal) firstArg).getValue() instanceof String) {
+            // Format: implementation "commons-lang:commons-lang:2.6"
+            String gav = ((String) ((J.Literal) firstArg).getValue());
+            String[] parts = gav.split(":");
+            String group = parts[0];
+            String artifact = parts[1];
+            String version = parts.length > 2 ? parts[2] : null;
+            return new ResolvedGroupArtifactVersion(group, artifact, version);
+        } else if (firstArg instanceof J.MethodInvocation) {
+            String artifact = "";
+            try {
+                artifact = ((J.Literal) ((J.MethodInvocation) firstArg).getArguments().get(0)).getValue().toString();
+                return new ResolvedGroupArtifactVersion(null, artifact, null);
+            } catch (Exception e) {
+                // todo log it
+                return null;
+            }
+        }
+        return null;
+    }
+
+    public static class Matcher extends SimpleTraitMatcher<GradleDependency> {
```

Comment:
I think we should have the Gradle-specific implementation of the trait matched built and then use that for confirming the dependency configurations rather than trying to guess at them.